PCSX2/pcsx2
# Build: Unify shared libraries across platforms

### Description of Changes

Use DLLs/shared libraries for:
 - libjpeg
 - libpng
 - libwebp
 - lz4
 - SDL
 - zlib
 - zstd

### Rationale behind Changes

 - Consistency across platforms.
 - ~1MB smaller exe.
 - Slightly quicker LTO times because less codegen.
 - Faster builds because more precompiled deps.
 - Less code.
 - Fewer goshdarn submodules.
 - Enabling me to switch imgui to freetype without bloating the binary further.

### Suggested Testing Steps

 - Make sure all platforms still run as expected.
 - Make sure eyetoy emulation still works, since I swapped jpgd for libjpeg.
